### ISSUE TITLE
Cannot reproduce results of TFLite INT8 model

### ISSUE THREAD

**Issue by adamp87** [ultralytics/ultralytics#7371]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Export

## Problem

When exporting and validating the TFLite INT8 model, the results claimed in PR #1695 cannot be reproduced. I have tested in Colab the latest ultralytics release and also executed @motokimura repo, both ending in same results.

motokimura repo executed as:
```
!git clone https://github.com/motokimura/ultralytics.git
%cd ultralytics

!git checkout -b fix/tflite_integer_quant origin/fix/tflite_integer_quant

!python setup.py install

!yolo export model=yolov8n.pt format=tflite int8
!yolo val model=yolov8n_saved_model/yolov8n_full_integer_quant.tflite data=coco128.yaml
```

Reproduced results compared to claimed results:
| Model & DSet | mAP50 | mAP50-95 | Comment |
| ------------- | ------------- | ------------- | ------------- |
| TFLite INT8 (yolo8n) Reproduce & COCO128 | 37.8 | 16.7 | |
| TFLite INT8 (yolo8n)  Reproduce & COCO |  25.4  |  8.68  | |
| TFLite INT8 (yolo8n)  Claimed & COCO128 |  52.4 | 35.8  | https://github.com/ultralytics/ultralytics/pull/1695 |
| TFLite INT8 (yolo8n)  Claimed & COCO |  46.4  |  28.7  | https://github.com/ultralytics/ultralytics/pull/1695#issuecomment-1518653422 |

## Solution

I would propose a solution by not normalizing the bounding boxes, but rather working with normalized constant factors for calculating the bounding boxes. I believe there must be a change in onnx2tf, but could not reproduce the source of the bug.

The proposed solution improves as follows:

1. Fix mentioned bug in repo
2. Improve accuracy for exported TFLite INT8 models compared to the claimed results
3. Working quantized model for large input image sizes e.g. 1280

### Details of solution

Normalize constants, not normalize results.

The output of the DFL block is bounded by [0, c1-1], where c1 is 16 by default.

- The softmax layer is applied on dim=1, therefore the c1 channel is bounded by [0, 1] and sums to 1.
- The Conv2D weights are set by the arrange function from [0, c1-1]

The standard output is computed by the dist2box function using the anchor values, and it is multiplied with stride. It is bounded by [0, img_size].  For the quantized TFLite models this output is normalized with img_size. This is an operation with larger values compared to the score outputs, resulting in higher numerical instability. See image 1.

The suggested solution is to normalize the anchor and stride values, which can be constants for a static model, so operations can be done in lower ranges. The output of the DFL block can be multiplied by the constants and the dist2box function can use the normalized anchor values, resulting in a more numerical stable calculation. See image 2.

## Improved results:

| Model & DSet | mAP50 | mAP50-95 | Comment |
| ------------- | ------------- | ------------- | ------------- |
| TFLite INT8 (yolo8n)  Improved & COCO128 | 55.2 | 39.2 | |
| **TFLite INT8 (yolo8n)  Improved & COCO** |  **48.4** | **32.7**  | |
| TFLite INT8 (yolo8n)  Claimed & COCO |  46.4  |  28.7  | https://github.com/ultralytics/ultralytics/pull/1695#issuecomment-1518653422 |
| yolo8n.pt & COCO | 52.5 | 37.3  | https://github.com/ultralytics/ultralytics/pull/1695#issuecomment-1518653422 |
| TFLite INT8 (yolo8n) without quantizing head + ReLU6 & COCO | ? | 34.7  | https://github.com/ultralytics/ultralytics/pull/1695#issuecomment-1652904495 https://github.com/ultralytics/ultralytics/pull/4495 | 

## Figures

![image](https://github.com/ultralytics/ultralytics/assets/7806910/a8c6c66f-4989-451d-9a3f-f35235f509de)
Image 1, showing large difference between quantization factors for current solution.

![image](https://github.com/ultralytics/ultralytics/assets/7806910/952aae84-9942-43bf-b1fa-a9b859a73328)
Image 2, showing small difference between quantization factors for proposed solution.

Complete SVG for both models
[yolov8n_full_integer_quant_svg.tflite.zip](https://github.com/ultralytics/ultralytics/files/13853500/yolov8n_full_integer_quant_svg.tflite.zip)

### Environment

Ultralytics YOLOv8.0.236 🚀 Python-3.10.12 torch-2.1.0+cu121 CPU (Intel Xeon 2.20GHz)
Setup complete ✅ (2 CPUs, 12.7 GB RAM, 26.3/107.7 GB disk)

OS                  Linux-6.1.58+-x86_64-with-glibc2.35
Environment         Colab
Python              3.10.12
Install             pip
RAM                 12.67 GB
CPU                 Intel Xeon 2.20GHz
CUDA                None

matplotlib          ✅ 3.7.1>=3.3.0
numpy               ✅ 1.23.5>=1.22.2
opencv-python       ✅ 4.8.0.76>=4.6.0
pillow              ✅ 9.4.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.11.4>=1.4.1
torch               ✅ 2.1.0+cu121>=1.8.0
torchvision         ✅ 0.16.0+cu121>=0.9.0
tqdm                ✅ 4.66.1>=4.64.0
psutil              ✅ 5.9.5
py-cpuinfo          ✅ 9.0.0
thop                ✅ 0.1.1-2209072238>=0.1.1
pandas              ✅ 1.5.3>=1.1.4
seaborn             ✅ 0.12.2>=0.11.0

### Minimal Reproducible Example

!yolo export model=yolov8n.pt data=coco128.yaml format=tflite int8
!yolo val model=yolov8n_saved_model/yolov8n_full_integer_quant.tflite data=coco128.yaml

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by adamp87** [comment]
I close this issue as #7372 has been merged.

**Comment by glenn-jocher** [comment]
@adamp87 issue closed as the related PR #7372 has been merged. Thank you for your contribution! 🎉 If you have further questions or encounter any more issues, feel free to open a new issue. Happy coding!
